### DISCUSSION TITLE
adevice_list.lua似乎不能正确识别默认指定的音频设备

### DISCUSSION THREAD

**Discussion by redomCL** [dyphire/mpv-config#111] [Q&A] [answered]
大佬，adevice_list.lua似乎不能正确识别默认指定的音频设备。如图：
当通过 audio-device=”wasapi/{12aafe68-6b26-44c2-8a71-d0cc5737bef1}“指定设备后，uosc可以正确识别，adevice_list依然指向auto：
![QQ截图20240818022209](https://github.com/user-attachments/assets/f20480d1-b8ce-4687-82fe-1af96b9d1da1)

**Comment by dyphire** [comment, marked as answer]
由提交 https://github.com/dyphire/mpv-config/commit/b0abfbb 修复

**Comment by dyphire** [comment]
又想了想，之前几个 list 菜单脚本的实现都不是很好，在提交 https://github.com/dyphire/mpv-config/commit/8470f16 中重构了

**Reply by redomCL** [reply]
track_list还重构吗？目前用着感觉很好了

**Reply by dyphire** [reply]
就是照着 track_list 重构的

**Reply by redomCL** [reply]
妥妥的

**Reply by dyphire** [reply]
> track_list还重构吗？目前用着感觉很好了

还是把 track-list.lua 重构了下，见 1e5c258 。之前的代码还是不够优雅（
